### PR TITLE
Fix the output of rbd info in expand

### LINKED ISSUE #88643: Use intree-rbd to expand pvc failed
**What happened**:

When I use intree-rbd to expand a pvc from 20Gi to 30Gi, it has been in the Resizing state for a long time.

```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    pv.kubernetes.io/bind-completed: "yes"
    pv.kubernetes.io/bound-by-controller: "yes"
    volume.beta.kubernetes.io/storage-provisioner: kubernetes.io/rbd
  creationTimestamp: "2020-02-28T02:36:26Z"
  finalizers:
  - kubernetes.io/pvc-protection
  name: intree-rbd-pvc
  namespace: default
  resourceVersion: "113553071"
  selfLink: /api/v1/namespaces/default/persistentvolumeclaims/intree-rbd-pvc
  uid: 1e3887cd-59d3-11ea-b7ad-eeeeeeeeeeee
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 30Gi
  storageClassName: intree-rbd-sc
  volumeMode: Filesystem
  volumeName: pvc-1e3887cd-59d3-11ea-b7ad-eeeeeeeeeeee
status:
  accessModes:
  - ReadWriteOnce
  capacity:
    storage: 20Gi
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2020-02-28T02:47:59Z"
    status: "True"
    type: Resizing
  phase: Bound
```

And kube-controller-manager reported this error：

```
E0228 02:48:02.557189       1 nestedpendingoperations.go:267] Operation for "\"1e3887cd-59d3-11ea-b7ad-eeeeeeeeeeee\"" failed. No retries permitted until 2020-02-28 02:48:03.05705946 +0000 UTC m=+327.117830161 (durationBeforeRetry 500ms). Error: "error expanding volume \"default/intree-rbd-pvc\" of plugin \"kubernetes.io/rbd\": rbd info failed, error: parse rbd info output failed: 2020-02-28 02:47:59.305808 7f769ab000 -1 did not load config file, using default settings.\n2020-02-28 02:47:59.320080 7f769ab000 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin: (2) No such file or directory\n2020-02-28 02:47:59.321712 7f747ee850  0 -- :/243049777 >> 172.16.2.105:6789/0 pipe(0x1213c9fa0 sd=3 :0 s=1 pgs=0 cs=0 l=1 c=0x1213c7e00).fault\n{\"name\":\"kubernetes-dynamic-pvc-11614948-59d4-11ea-a0dd-eeeeeeeeeeee\",\"size\":21474836480,\"objects\":5120,\"order\":22,\"object_size\":4194304,\"block_name_prefix\":\"rbd_data.b372176b8b4567\",\"format\":2,\"features\":[\"layering\"],\"flags\":[]}, invalid character '-' after top-level value"
I0228 02:48:02.557190       1 event.go:209] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"intree-rbd-pvc", UID:"1e3887cd-59d3-11ea-b7ad-eeeeeeeeeeee", APIVersion:"v1", ResourceVersion:"113553071", FieldPath:""}): type: 'Warning' reason: 'VolumeResizeFailed' error expanding volume "default/intree-rbd-pvc" of plugin "kubernetes.io/rbd": rbd info failed, error: parse rbd info output failed: 2020-02-28 02:47:59.305808 7f769ab000 -1 did not load config file, using default settings.
2020-02-28 02:47:59.320080 7f769ab000 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin: (2) No such file or directory
2020-02-28 02:47:59.321712 7f747ee850  0 -- :/243049777 >> 172.16.2.105:6789/0 pipe(0x1213c9fa0 sd=3 :0 s=1 pgs=0 cs=0 l=1 c=0x1213c7e00).fault
{"name":"kubernetes-dynamic-pvc-11614948-59d4-11ea-a0dd-eeeeeeeeeeee","size":21474836480,"objects":5120,"order":22,"object_size":4194304,"block_name_prefix":"rbd_data.b372176b8b4567","format":2,"features":["layering"],"flags":[]}, invalid character '-' after top-level value
```

The most important msg is:

```
rbd info failed, error: parse rbd info output failed
```

I found this err:
https://github.com/kubernetes/kubernetes/blob/79377454e8dfa4c9c96f786c2f34e2a6230f7139/pkg/volume/rbd/rbd_util.go#L735

After I read this part of the code, I realized that the output of command "rbd info" is not a struct.

**What you expected to happen**:

Pvc expand successful.

### PR DESCRIPTION
**Pull request by huchengze**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Fix the output of rbd info in expand。

**Which issue(s) this PR fixes**:

Fixes #88643

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by huchengze** [comment]
@jsafrane

**Comment by php-coder** [comment]
@huchengze Thank you for the contribution!

It would be great to also have a test for that case. Could you provide it?

**Comment by huchengze** [comment]
> @huchengze Thank you for the contribution!
> 
> It would be great to also have a test for that case. Could you provide it?

Sorry, I'm a little confused. Do you mean that I should add some test cases in:
https://github.com/kubernetes/kubernetes/blob/25d241eda2358e616752751713437f9e9aa8e904/pkg/volume/rbd/rbd_test.go#L687

**Comment by jsafrane** [comment]
Yes, please add more tests (incl. error cases) to `TestGetRbdImageSize()`

**Comment by jsafrane** [comment]
pull-kubernetes-verify failed with:
```
# k8s.io/kubernetes/pkg/volume/rbd
vet: pkg/volume/rbd/rbd_test.go:701:32: cannot use []byte(c.Output) (value of type []byte) as []string value in argument to getRbdImageSize
make[1]: *** [Makefile:386: vet] Error 1
```

**Comment by huchengze** [comment]
@jsafrane Sorry, I'm a little busy these days, I will fix it this weekend.

File: pkg/volume/rbd/rbd_util.go [hunk 1]
```diff
@@ -724,17 +724,26 @@ func (util *RBDUtil) rbdInfo(b *rbdMounter) (int, error) {
...
+func getRbdImageSize(outputs []string) (int, error) {
+	var err error
 	info := struct {
 		Size int64 `json:"size"`
 	}{}
-	if err := json.Unmarshal(output, &info); err != nil {
-		return 0, fmt.Errorf("parse rbd info output failed: %s, %v", string(output), err)
+
+	for _, output := range outputs {
+		if err = json.Unmarshal([]byte(output), &info); err != nil {
```
Comment (juliantaylor) [inline pkg/volume/rbd/rbd_util.go]: a better fix is to only use the stdout of `b.exec.Command` in line 707
rbd info prints informational data on stderr, but with --format=json as used the json lands on stdout which then can easily be parsed

Comment (sfudeus) [reply pkg/volume/rbd/rbd_util.go]: I agree - it looks error prone to me to just consume lines with parseable json and throw away the other ones without notice. What if rbd info emits multiline json?

**Pull request closed without merging**